tests/test_edge_cases.py

```diff
@@ -570,6 +590,8 @@ class Model(BaseModel):
 
     m = Model(e='e', f=SubModel(c='foo', d=[SubSubModel(a='a', b='b'), SubSubModel(a='c', b='e')]))
 
+    # TODO: Do we still want to support this wrapping behavior on exclude?
+    #   If so, I am happy to implement it in pydantic_core. I got a good ways through but abandoned it until we discuss.
     assert m.model_dump(exclude={'f': {'c': ..., 'd': {-1: {'a'}}}}) == {
```

Comment:
Other spot where I wasn't sure if the goal was to support this behavior or not..